### PR TITLE
Refactor/event listener clear

### PR DESCRIPTION
**Pull request by thetarnav**
Related issue: #103 

Affected packages:
- [x] `event-listener` - [readme](https://github.com/solidjs-community/solid-primitives/tree/refactor/event-listener-clear/packages/event-listener#readme)

   - Remove `createEventListenerBus`, `createEventListenerStore` & `eventListenerMap`
   - Add `makeEventListener`, `makeEventListenerStack`
   - Remove clear() functions from reactive primitives.

- [x] `active-element` - [readme](https://github.com/solidjs-community/solid-primitives/tree/refactor/event-listener-clear/packages/active-element#readme)

   - Moved to stage 3
   - Renamed `createIsElementActive` to `createFocusSignal` nad `isElementActive` directive to `focus`.
   - Add `makeActiveElementListener` & `makeFocusListener` non-reactive primitives.
   - Removed clear() functions from reactive primitives.

- [x] `mouse` - [readme](https://github.com/solidjs-community/solid-primitives/tree/refactor/event-listener-clear/packages/mouse#readme)

   - moved to stage 3
   - Removed `createMouseInElement`, `createMouseOnScreen`
   - Renamed:
	- `posRelativeToElement` -> `getPositionToElement`
	- `posRelativeToScreen` -> `getPositionToScreen`
	- `createMouseToElement` -> `createPositionToElement`
   - Added `makeMousePositionListener`, `makeMouseInsideListener` and `getPositionInElement`
   - Removed clear() and update() functions from reactive primitives. createPositionToElement now only takes accessor position.

- [x] `connectivity` - [readme](https://github.com/solidjs-community/solid-primitives/tree/refactor/event-listener-clear/packages/connectivity#readme)

   - moved to stage 2
   - Add `makeConnectivityListener` and `useConnectivitySignal`

- [x] `date` - [readme](https://github.com/solidjs-community/solid-primitives/tree/refactor/event-listener-clear/packages/date#solid-primitivesdate)

   - Remove `createTime`, use memo, and timer packages to reuse primitives

- `media` - [readme](https://github.com/solidjs-community/solid-primitives/tree/main/packages/media#readme)
   - refactor, add `makeMediaQueryListener`

### CODE REVIEW

File: packages/active-element/README.md [hunk 1]
```diff
@@ -110,4 +153,12 @@ Updated event listener and util dependencies.
 
 Updated to Solid 1.3
 
+2.0.0 - **stage-3**
+
+Renamed `createIsElementActive` to `createFocusSignal` nad `isElementActive` directive to `focus`.
```
Comment (bgoscinski) [inline packages/active-element/README.md:158]:
```suggestion
Renamed `createIsElementActive` to `createFocusSignal` and `isElementActive` directive to `focus`.
```

Comment (thetarnav) [reply packages/active-element/README.md]: Thank you :)

File: packages/mouse/test/index.test.ts [hunk 2]
```diff
@@ -45,36 +36,46 @@ const mte = suite("createMouseToElement");
...
-    assert.type(isInside, "function");
-    // isInside is calculated with position and element bounds,
-    // so x is greater or equal to 0
-    assert.is(isInside(), true);
-
-    assert.type(update, "function");
+    const [pos, setPos] = createStaticStore({ x: 0, y: 0 });
+    const relative = createPositionToElement(el, () => pos);
+
+    assert.is(relative.x, 0);
```
Comment (atk) [inline packages/mouse/test/index.test.ts:42]: You know that you can use `assert.equal` to compare objects?

**Review by atk — APPROVED** [review]

File: packages/active-element/src/index.ts [hunk 3]
```diff
@@ -1,83 +1,101 @@
...
+ * const [isFocused, setIsFocused] = createSignal(false)
+ * const clear = makeFocusListener(focused => setIsFocused(focused));
+ * // remove listeners (happens also on cleanup)
+ * clear();
+ */
+export function makeFocusListener(
+  target: Element,
+  callback: (isActive: boolean) => void
+): VoidFunction {
+  const clear1 = makeEventListener(target, "blur", () => callback(false), true);
```
[the remark this answers is not in the document]
Comment (thetarnav) [reply packages/active-element/src/index.ts:67]: Add a comment to the `makeEventListener` or `makeFocusListener`? or both?
For `makeEventListener` it is just mirroring the Browser API, and is already linking to [MDN](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener)
But for `makeFocusListener` it makes sense to mention it, or even allow controlling it via optional third argument.

File: packages/active-element/package.json [hunk 4]
```diff
@@ -11,10 +11,10 @@
   },
   "primitive": {
     "name": "active-element",
-    "stage": 2,
+    "stage": 3,
     "list": [
       "createActiveElement",
-      "createIsElementActive"
+      "createFocusSignal"
```
[the remark this answers is not in the document]
Comment (thetarnav) [reply packages/active-element/package.json:17]: Yeah I honestly don't know. There are:
- reactive primitives (createActiveElement, createMousePosition)
- non-reactive primitives (makeFocusListener, makeTimer, makeMousePositionListener)
- "shared primitives" (useMousePosition, useLocationState, useDevices)
- calculations (getPositionToElement, getTimeDifference)
- directives (focus, pointerHover, unmount)
- components (WindowEventListener, Range, Key)
- possibly other misc utility functions

Which of these should be listed?
All of them? Packages like [mouse](https://github.com/solidjs-community/solid-primitives/tree/refactor/event-listener-clear/packages/mouse), [event-listener](https://github.com/solidjs-community/solid-primitives/tree/refactor/event-listener-clear/packages/event-listener), [date](https://github.com/solidjs-community/solid-primitives/tree/refactor/event-listener-clear/packages/date) do export a lot of separate functions
Some of them? Then we are always gonna be omitting some of the APIs...

@davedbase some opinions on this?

File: packages/connectivity/tsconfig.json [hunk 5]
```diff
@@ -1,7 +1,9 @@
 {
   "extends": "../../tsconfig.json",
   "include": [
-    "./src"
+    "./src",
+    "./test",
```
[the remark this answers is not in the document]
Comment (thetarnav) [reply packages/connectivity/tsconfig.json:5]: It is solely for type checking, we don't use tsconfig for building at all.

File: packages/active-element/src/index.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (atk) [reply packages/active-element/src/index.ts:67]: one comment should suffice.

**Pull request merged**